src/net_processing.cpp

```diff
@@ -2586,7 +2585,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         bool fAnnounceUsingCMPCTBLOCK = false;
         uint64_t nCMPCTBLOCKVersion = 0;
         vRecv >> fAnnounceUsingCMPCTBLOCK >> nCMPCTBLOCKVersion;
-        if (nCMPCTBLOCKVersion == 1 || ((pfrom.GetLocalServices() & NODE_WITNESS) && nCMPCTBLOCKVersion == 2)) {
+        if (nCMPCTBLOCKVersion == 1 || nCMPCTBLOCKVersion == 2) {
```

Comment:
It [think](https://developer.bitcoin.org/reference/p2p_networking.html#sendcmpct) the only allowed values are 1 or 2. Can this conditional be removed altogether? Or perhaps replaced with `if (0 < nCMPCTBLOCKVersion < 3)` ?

Reply:
Yes, the only permitted values for nCMPCTBLOCKVersion are `1` and `2`, which is why this conditional is here. We want to ignore any `sendcmpct` message received where nCMPCTBLOCKVersion is not 1 or 2.

`if (0 < nCMPCTBLOCKVersion < 3)` does not do what you I think you expect it to do.

Reply:
Ok, I've leave the conditional in there then! :)